### PR TITLE
Add warning banner for audit access expiration

### PR DESCRIPTION
**Pull request by stvstnfrd**
to inform users of deadline and prompt them to upgrade to the verified
access track.

This depends on:
- https://github.com/edx/edx-platform/pull/23709

### CODE REVIEW

File: src/access-expiration-alert/AccessExpirationAlert.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+import Alert from '../user-messages/Alert';
+
+function AccessExpirationAlert(props) {
+  const rawHtml = props.rawHtml;
+  return rawHtml && (
+    <Alert type="info">
+      <div dangerouslySetInnerHTML={{ __html: rawHtml }} />
```
Comment (abutterworth) [inline src/access-expiration-alert/AccessExpirationAlert.jsx:12]: Will TNL-7185 allow us to avoid doing this?

**Review by abutterworth — APPROVED** [review]
A couple lint errors.
LGTM

File: src/access-expiration-alert/AccessExpirationAlert.jsx (hunk 1 above)
Comment (stvstnfrd) [reply src/access-expiration-alert/AccessExpirationAlert.jsx:12]: Yup!

**Comment by stvstnfrd** [comment]
Linting issues addressed and build is green.

File: src/access-expiration-alert/hooks.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { useContext, useState, useEffect } from 'react';
+import UserMessagesContext from '../user-messages/UserMessagesContext';
+import { useModel } from '../model-store';
+
+export function useAccessExpirationAlert(courseId) {
+  const course = useModel('courses', courseId);
+  const { add, remove } = useContext(UserMessagesContext);
+  const [alertId, setAlertId] = useState(null);
+  const expiredMessage = (course && course.courseExpiredMessage) || null;
+  useEffect(() => {
```
Comment (davidjoy) [inline src/access-expiration-alert/hooks.js:11]: I'd love to clean this stuff up across these various custom alerts - it's a hell of a piece of code.  (nothing to do for this PR, just adding this for posterity)

**Review by davidjoy — APPROVED** [review]

**Pull request merged**